x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -25,12 +27,31 @@ export const useInitSourcerer = (
   const initialTimelineSourcerer = useRef(true);
   const initialDetectionSourcerer = useRef(true);
   const { loading: loadingSignalIndex, isSignalIndexExists, signalIndexName } = useUserInfo();
-  const getConfigIndexPatternsSelector = useMemo(
-    () => sourcererSelectors.configIndexPatternsSelector(),
+  const getDefaultIndexPatternSelector = useMemo(
+    () => sourcererSelectors.defaultIndexPatternSelector(),
     []
   );
-  const ConfigIndexPatterns = useDeepEqualSelector(getConfigIndexPatternsSelector);
+  const defaultIndexPattern = useDeepEqualSelector(getDefaultIndexPatternSelector);
+  const { addError } = useAppToasts();
 
+  useEffect(() => {
+    const { id, ...rest } = defaultIndexPattern;
+    if (id === null) {
+      // if id is null, rest is error
+      addError(rest, {
```

Comment:
I think it might be worth it to set an explicit `error` param rather than relying on rest here to keep separation of concerns?

Reply:
im going to iterate on this in follow up PRs but will keep this comment open for a reminder